walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/servlets/FindMeetingQuery.java

```diff
@@ -0,0 +1,152 @@
+                                    removeList.add(openTime);
+                                    TimeRange overlapTime = TimeRange.fromStartEnd(whenEvent.end(),openTime.end(),false);
+                                    if (compareTimeRangeWithMeeting(meetingDuration, overlapTime.duration()))
+                                    {
+                                        addList.add(overlapTime);
+                                    }
+                                }
+                            }
+                        }    
+                    }
+                    availableTime.removeAll(removeList);
+                    availableTime.addAll(addList);
+                    finalTimeList = availableTime; 
+                }  
+                else if (optionalAttendees.isEmpty())
+                {
+                    
+                    finalTimeList = Arrays.asList(TimeRange.WHOLE_DAY);
+                } 
+                //Checks if optional attendees can attend open times, but will not affect times made for mandatory attendees
+                if (checkAttendees(eventAttendees, optionalAttendees) && !meetingAttendees.isEmpty())
+                {
+                    Collection<TimeRange> timeWithOptional = new ArrayList<TimeRange>(); 
+                    int count = 0;
+                    for(TimeRange currentOpenTime: availableTime)
+                    {
+                        if(!currentOpenTime.overlaps(whenEvent))
+                        {
+                            count++;
+                            timeWithOptional.add(currentOpenTime);
+
+                        }
+                    }
+                    if (count !=0)
+                    {
+                        finalTimeList = timeWithOptional;
+                    }
+                } 
+            }
+        }
+        return finalTimeList;
+    }
+
+    /** Add open slot for potential meeting times before and after the event*/
+    public void addFrontAndBackTimesToList(long meetingDuration, TimeRange frontTime,TimeRange backTime, Collection<TimeRange> timeRangeList)
+    {
+        if (compareTimeRangeWithMeeting(meetingDuration, frontTime.duration()))
+        {
+            timeRangeList.add(frontTime);
+        }
+        if (compareTimeRangeWithMeeting(meetingDuration, backTime.duration()))
+        {
+            timeRangeList.add(backTime);
+            
+        }
+    }
+
+    /** Check if event attendees are on the meeting or optional attendees list*/
+    public boolean checkAttendees(Set<String> eventAttendees, Collection<String> checkAttendees)
+    {
+        for(String name: eventAttendees)
+        {
+            if (checkAttendees.contains(name))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /** Compare a potential time slot to make sure the meeting duration can fit */
+    public boolean compareTimeRangeWithMeeting(long meetingDuration, long openSlotTime)
```

Comment:
For this method, I think it would be more clear not to have it, and just put the actual logic in its place.

For example, instead of

```
if (compareTimeRangeWithMeeting(meetingDuration, overlapTime.duration()))
```

You would have

```
if (meetingDuration <= overlapTime.duration())
```